app/src/main/java/com/randomappsinc/studentpicker/common/VoiceRecognizerDialogFragment.java

```diff
@@ -0,0 +1,161 @@
+package com.randomappsinc.studentpicker.common;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.speech.RecognitionListener;
+import android.speech.RecognizerIntent;
+import android.speech.SpeechRecognizer;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+
+import com.joanzapata.iconify.widget.IconTextView;
+import com.randomappsinc.studentpicker.R;
+
+import java.util.ArrayList;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+public class VoiceRecognizerDialogFragment extends DialogFragment implements RecognitionListener {
+
+    public interface VoiceRecognizerInterface {
```

Comment:
nit: This is more in line with the rest of the app. I'm not a fan of putting the entity type (in this case "interface") into the name, since it's redundant information.
```suggestion
    public interface Listener {
```